### PR TITLE
Started addition of new exams

### PR DESCRIPTION
**Pull request by conleyrj**
Started adding exams to the list. This works but there is no verification. No method of deletion yet either and these are deleted after refresh. Only using Create Exam button in top right of screen instead of having a new page on the sidebar. A pop-up window comes up to enter info

Also attempted to start having some page memory to make sidebar usable, but am having trouble implementing it in certain spots.

### CODE REVIEW

File: app/src/components/dashboard/index.js [hunk 1]
```diff
@@ -32,7 +72,39 @@ const Page = () => {
...
+                      To create a new exam, please enter a title for the exam and the name of the class the exam is being created for.
+                    </DialogContentText>
+                     <TextField
+                      placeholder="Enter the name of the Exam here"
+                      margin="dense"
+                      id="examName"
+                      label="Exam Name"
+                      fullWidth
+                      variant="outlined"
+                      value={examName}
```
Comment (melaniemai) [inline app/src/components/dashboard/index.js:89]: I think with the value set to `examName`, if a user 
1. enters in a name
2. clicks outside the dialog box
3. then clicks the 'Create Exam' button again
what they previously typed in stays. Do we want to clear the text input?

File: app/src/components/dashboard/index.js [hunk 2]
```diff
@@ -32,7 +72,39 @@ const Page = () => {
...
+                      label="Class Name"
+                      fullWidth
+                      variant="outlined"
+                      value={className}
+                      onChange={e => {setClassName(e.target.value)}}
+                    />
+                  </DialogContent>
+                  <DialogActions>
+                    <Button onClick={handleClose}>Cancel</Button>
+                    <Button onClick={handleCreate}>Create</Button>
```
Comment (melaniemai) [inline app/src/components/dashboard/index.js:105]: This is minor and NOT necessary to fix ASAP but could we add a 
`variant="contained" color="success"` to the "Create" button and 
`variant="contained" color="error"` to the "Cancel"?

**Review by melaniemai — COMMENTED** [review]
ry: "im not good at frontend"
me: "it was all a lie this is beautiful"

File: app/src/components/dashboard/index.js (hunk 1 above)
Comment (conleyrj) [reply app/src/components/dashboard/index.js:89]: That's a good point, I will move the lines in the create action that reset the values to be in the close action so they happen whenever the dialog is closed

File: app/src/components/dashboard/index.js (hunk 2 above)
Comment (conleyrj) [reply app/src/components/dashboard/index.js:105]: sure, easy addition

File: app/src/components/dashboard/index.js [hunk 3]
```diff
@@ -1,14 +1,54 @@
...
 
+// Used for creating uuid for each exam that is added
+// TODO is there a different way we want to go about this?
+const { v4: uuidv4 } = require('uuid');
+
 const Page = () => {
   const theme = useTheme()
+  const examList = usePaperStore(store => store.examList)
+  const setExamList = usePaperStore(store => store.setExamList)
+
```
Comment (melaniemai) [inline app/src/components/dashboard/index.js:20]: Don't we need to set a `setDisplayPage` somewhere here? I guess we'd have a better idea if pages are properly switching/working if we set up the Settings page and all that 🤔

File: app/src/components/dashboard/index.js [hunk 4]
```diff
@@ -32,7 +72,39 @@ const Page = () => {
...
+                      fullWidth
+                      variant="outlined"
+                      value={className}
+                      onChange={e => {setClassName(e.target.value)}}
+                    />
+                  </DialogContent>
+                  <DialogActions>
+                    <Button variant="contained" color="error" onClick={handleClose}>Cancel</Button>
+                    <Button variant="contained" color="success" onClick={handleCreate}>Create</Button>
+                  </DialogActions>
```
Comment (melaniemai) [inline app/src/components/dashboard/index.js:106]: Actually, could we change the buttons so they're something like this? 
```
<Button sx={{ color: "#8F90A6"}} onClick={handleClose}>Cancel</Button>
<Button variant="contained" color="primary" onClick={handleCreate}>Create</Button>
```
And could we also fix the padding for the `DialogActions`? I think the button group would look better with `20px 24px`? Just so it matches the rest of the dialog box 😌

**Pull request closed without merging**